### PR TITLE
Remove idle ShellError exception and get_exception() method from nxos modules

### LINKED ISSUE #23890: nxos_interface crashes during int Vlan creation
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

nxos_interface

##### ANSIBLE VERSION

```
ubuntu@ubuntu-xenial:/vagrant/ansible-nxos$ ansible --version
ansible 2.3.0.0
  config file = /vagrant/ansible-nxos/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```

##### OS / ENVIRONMENT

Ubuntu

##### SUMMARY

Playbook crashes during  vlan creation task

##### STEPS TO REPRODUCE

```yaml

  tasks:
    - name: Create interface
      nxos_interface:
        interface: Vlan50
        description: "Description"


```

##### EXPECTED RESULTS

The Vlan was created but the play crashed.

##### ACTUAL RESULTS

An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NameError: global name 'ShellError' is not defined
fatal: [nxos2]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_dex0aa/ansible_module_nxos_interface.py\", line 763, in <module>\n    main()\n  File \"/tmp/ansible_dex0aa/ansible_module_
nxos_interface.py\", line 675, in main\n    existing, is_default = smart_existing(module, intf_type, normalized_interface)\n  File \"/tmp/ansible_dex0aa/ansible_module_nxos_interface.py\", line 559, in smart_existing\n    existing = get_interface(normalized_in
terface, module)\n  File \"/tmp/ansible_dex0aa/ansible_module_nxos_interface.py\", line 326, in get_interface\n    attributes = get_manual_interface_attributes(intf, module)\n  File \"/tmp/ansible_dex0aa/ansible_module_nxos_interface.py\", line 248, in get_man
ual_interface_attributes\n    except (IndexError, ShellError):\nNameError: global name 'ShellError' is not defined\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 0}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NameError: global name 'ShellError' is not defined

---

It looks like exception ShellError is not defined inside the module
The line 588 with "body = response" should be probably deleted

```

```

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

##### SUMMARY

Remove idle `ShellError` exception and `get_exception()` method from nxos modules.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

- modules/network/nxos/nxos_bgp_af.py
- modules/network/nxos/nxos_bgp_neighbor.py
- modules/network/nxos/nxos_bgp_neighbor_af.py
- modules/network/nxos/nxos_evpn_vni.py
- modules/network/nxos/nxos_file_copy.py
- modules/network/nxos/nxos_igmp.py
- modules/network/nxos/nxos_interface_ospf.py
- modules/network/nxos/nxos_ntp_auth.py
- modules/network/nxos/nxos_ospf.py
- modules/network/nxos/nxos_ospf_vrf.py
- modules/network/nxos/nxos_pim.py
- modules/network/nxos/nxos_pim_rp_address.py
- modules/network/nxos/nxos_smu.py
- modules/network/nxos/nxos_vrf_af.py
- modules/network/nxos/nxos_vxlan_vtep.py

##### ANSIBLE VERSION

```
devel
```

### CODE REVIEW

**Review by gundalow — APPROVED** [review]

**Review by Qalthos — CHANGES_REQUESTED** [review]
So none of these modules actually need ShellError... it's an old carryover from pre-2.3 days. What you can do instead is remove the `try [...] except ShellError` blocks around the code, as that code should not throw ShellErrors

**Comment by trishnaguha** [comment]
@Qalthos Let me clarify that sounds like refactor of `nxos_interface` for `2.3` right? 
If yes I will push refactor of `nxos_interface` here in this PR and exclude other modules from the PR.

File: lib/ansible/modules/network/nxos/nxos_evpn_vni.py [hunk 1]
```diff
@@ -131,6 +131,7 @@
 import re
 from ansible.module_utils.nxos import get_config, load_config, run_commands
 from ansible.module_utils.nxos import nxos_argument_spec, check_args
+from ansible.module_utils.shell import ShellError
```
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_evpn_vni.py]: Do we really still need this?

File: lib/ansible/modules/network/nxos/nxos_interface.py [hunk 2]
```diff
@@ -243,10 +243,7 @@ def get_manual_interface_attributes(interface, module):
 
     if get_interface_type(interface) == 'svi':
         command = 'show interface ' + interface
-        try:
-            body = execute_modified_show_for_cli_text(command, module)[0]
-        except (IndexError, ShellError):
```
Comment (Qalthos) [inline lib/ansible/modules/network/nxos/nxos_interface.py]: The IndexError seems like it might still be relevant, given the `[0]` above

**Review by Qalthos — COMMENTED** [review]
Other than stepping on the toes of my nxos_bgp PR, this looks good. Look into the comments I left and this should be good to go.

File: lib/ansible/modules/network/nxos/nxos_evpn_vni.py (hunk 1 above)
Comment (trishnaguha) [reply lib/ansible/modules/network/nxos/nxos_evpn_vni.py]: Looks like I missed it.

**Comment by trishnaguha** [comment]
It makes sense to remove `nxos_interface` from this PR as the nxos_interface refactoring PR in on the way https://github.com/ansible/ansible/pull/24008 which would this change as well.

**Review by Qalthos — APPROVED** [review]

**Comment by Qalthos** [comment]
shipit

**Review by Qalthos — CHANGES_REQUESTED** [review]
Er, actually, rename that commit if you are no longer fixing the mentioned issue, otherwise github will probably autoclose it

**Comment by trishnaguha** [comment]
@Qalthos Oh yeah right. Thanks.
FYI #24008 has closed the issue #23890 . I will merge this PR now.

**Comment by trishnaguha** [comment]
This should be cherry picked in `stable-2.3`.

**Pull request merged**